src/fairness/writer/data_writer_stdout.hpp

```diff
@@ -0,0 +1,72 @@
+/************************************************************\
+ * Copyright 2021 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+#include <sqlite3.h>
+#include <cerrno>
+
+#include "src/fairness/weighted_tree/weighted_walk.hpp"
+#include "src/fairness/writer/data_writer_base.hpp"
+
+using namespace Flux::accounting;
+
+namespace Flux {
+namespace writer {
+
+/*!  Base data writer class.
+ */
+class data_writer_stdout_t : public data_writer_base_t {
+public:
+    virtual ~data_writer_stdout_t () = default;
+
+    /*! Write fairshare values from a weighted tree.
+     *
+     * \param node          node of the weighted tree of the bank/user hierarchy
+     * \param indent        indent amount when printing out parsable hierarchy
+     * \param parsable      boolean determining parsable or pretty output
+     * \param delimiter     custom delimiter for output
+     */
+    int write_acct_info (std::shared_ptr<weighted_tree_node_t> node,
+                         const std::string& indent,
+                         bool parsable,
+                         const std::string& delimiter="|");
```

Comment:
The signature of this `write_acct_info` differs from the signature in the base class.  Might make sense to refactor this class so they are the same.  I'll defer to @dongahn on the best way to do this, but maybe push the `indent`, `parsable`, `delimiter` configuration into the constructor of the writer (so the constructors of the various classes will differ) but the write function has the same signature.

Reply:
Thanks for pointing this out - I just pushed up a change that I hope works better. I moved `indent`, `parsable`, and `delimiter` to be public members of the `data_writer_stdout` subclass and kept the signature the same as the base class. This allows me to access these public members of a `data_writer_stdout` subclass with `this->member`.

I also moved some of the helper functions to `data_writer_stdout.cpp`, so `flux_shares.cpp` now really only parses the command line arguments and then calls `write_acct_info ()` with an optional filename.